fastlane/lib/fastlane/actions/notarize.rb

```diff
@@ -123,6 +130,35 @@ def self.staple(package_path, verbose)
         )
       end
 
+      def self.with_notarize_authenticator(api_key_path)
+        if api_key_path
+          # From xcrun altool for --apiKey:
+          # This option will search the following directories in sequence for a private key file with the name of 'AuthKey_<api_key>.p8':  './private_keys', '~/private_keys', '~/.private_keys', and '~/.appstoreconnect/private_keys'.
+          api_key = Spaceship::ConnectAPI::Token.from_json_file(api_key_path)
+          api_key_folder_path = File.expand_path('~/.appstoreconnect/private_keys')
+          api_key_file_path = File.join(api_key_folder_path, "AuthKey_#{api_key.key_id}.p8")
+          directory_exists = File.directory?(api_key_folder_path)
+          file_exists = File.exist?(api_key_file_path)
+          begin
+            FileUtils.mkdir_p(api_key_folder_path) unless directory_exists
+            api_key.write_key_to_file(api_key_file_path) unless file_exists
```

Comment:
I got errors when this action tried to write the file. I don’t _think_ I have my API file formatted incorrectly 🤔 

I _was_ able to get this action working by manually moving my `.p8` into this directory. Are you able to verify this still works on your end? 🙃 

```sh
Failed to authenticate with errors: (
    "Error Domain=ITunesConnectionAuthenticationErrorDomain Code=-26000 \"*** Failed to generate token: Error Domain=CupertinoJWT.CupertinoJWTError Code=1 \"The .p8 string has invalid format.\"\" UserInfo={NSLocalizedRecoverySuggestion=*** Failed to generate token: Error Domain=CupertinoJWT.CupertinoJWTError Code=1 \"The .p8 string has invalid format.\", NSLocalizedDescription=*** Failed to generate token: Error Domain=CupertinoJWT.CupertinoJWTError Code=1 \"The .p8 string has invalid format.\", NSLocalizedFailureReason=App Store operation failed.}"
)
```